### PR TITLE
Updated Player::CurrentRoom for tinybrain people

### PR DESCRIPTION
**Pull request by Monaldcry7788**
*Done with the phone, I may have been wrong*

### CODE REVIEW

**Comment by iopietro** [comment]
Just add a setter to [Player::CurrentRoom](https://github.com/Exiled-Team/EXILED/blob/cd4b256834bb78fe4d1ce6f5d1a8fb2dbd2546ee/Exiled.API/Features/Player.cs#L810).

**Comment by iRebbok** [comment]
> Just add a setter to [Player::CurrentRoom](https://github.com/Exiled-Team/EXILED/blob/cd4b256834bb78fe4d1ce6f5d1a8fb2dbd2546ee/Exiled.API/Features/Player.cs#L810).

It'll force devs to find an instance of `Room`, could a new property called `CurrentRoomType` or something be good to address this?

**Review by Monaldcry7788 — COMMENTED** [review]
Updated Player::CurrentRoom for sets the current player's room

**Comment by Remuchi** [comment]
Setting player position to room.gameobject.position will not work properly for some rooms. For example, 939 chamber (you will spawn in air, fall down and die), 079 room (you spawn between walls), collapsed tunnels in entrance zone etc 
Don't know about surface

**Comment by iopietro** [comment]
> It'll force devs to find an instance of Room, could a new property called CurrentRoomType or something be good to address this?
Yes, you're right, what do you think about having bot `CurrentRoom` and `CurrentRoomType`? 
In my opinion the first one could still be useful for devs who search for rooms directly from `Map.Rooms`.

> Setting player position to room.gameobject.position will not work properly for some rooms. For example, 939 chamber (you will spawn in air, fall down and die), 079 room (you spawn between walls), collapsed tunnels in entrance zone etc
Don't know about surface.

@Monaldcry7788 Because of that, you should use `PlayerMovementSync::FindSafePosition`, which is a base game method.

**Comment by iRebbok** [comment]
> Yes, you're right, what do you think about having bot CurrentRoom and CurrentRoomType?
In my opinion the first one could still be useful for devs who search for rooms directly from Map.Rooms.

Sure!

**Comment by Monaldcry7788** [comment]
Updated

**Comment by 4310V343k** [comment]
Chill, chill! It's only one line!

<details>
<summary></summary>



</details>

**Comment by Monaldcry7788** [comment]
Removed white line at L1194

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -777,9 +777,28 @@ public string GroupName
...
+        /// Gets or sets the current room the player is in.
         /// </summary>
-        public Room CurrentRoom => Map.FindParentRoom(GameObject);
+        public Room CurrentRoom
+        {
+            get => Map.FindParentRoom(GameObject);
+            set
+            {
+                PlayerMovementSync.FindSafePosition(value.Position, out Vector3 safePos);
+                switch (value.Type)
```
Comment (joker-119) [inline Exiled.API/Features/Player.cs:788]: There needs to be more problematic rooms added to this check, for example: 173's room, 079's room, 939's room, etc. You also need to take into account that these rooms can be facing 4 different directions along the x axis, Vector3.back may not always be the safest direction for a particular room.

**Review by joker-119 — CHANGES_REQUESTED** [review]
More rooms need to be added to the switch check, for ex, 173's room will put you in a wall, 079's put you inside the cage, 939's puts you in the middle of the air, etc.

Also you need to keep in mind that all rooms can be facing 4 different directions along the x axis, Vector3.back (negative z) may not always be the direction of 'safety', I'd recommend using the room's transform instead of the static Vector3

**Pull request closed without merging**